pkg/platform/kube/logProxy/elastic/opensearch.go

```diff
@@ -48,23 +53,249 @@ func NewOpenSearchLogProxy(config *platformconfig.ElasticSearchConfig) (*OpenSea
+
+	if options.Size != 0 {
+		query["size"] = options.Size
+	}
+
+	if options.From != 0 {
+		query["from"] = options.From
+	}
+
+	// Add search_after if specified
+	if len(options.SearchAfter) > 0 {
+		query["search_after"] = options.SearchAfter
+	}
+
+	searchRequest, err := o.getSearchRequestFromQuery(query)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to create search request")
+	}
+
+	request, err := searchRequest.GetRequest()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to generate http request from search request")
+	}
+
+	// Perform search
+	res, err := o.client.Client.Perform(request)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to execute search")
+	}
+
+	return res.Body, nil
 }
 
-func (o *OpenSearchLogProxy) GetFunctionReplicas(ctx context.Context, options *logProxy.GetFunctionReplicaOptions) ([]string, error) {
-	return nil, nil
+func (o *OpenSearchLogProxy) getFunctionBaseSearchRequest(functionName string) map[string]interface{} {
+	return map[string]interface{}{
+		"query": map[string]interface{}{
+			"bool": map[string]interface{}{
+				"must": []interface{}{
+					map[string]interface{}{
+						"wildcard": map[string]interface{}{
+							"kubernetes.pod.name": fmt.Sprintf("nuclio-%s-*", functionName),
+						},
+					},
+					map[string]interface{}{
+						"query_string": map[string]interface{}{
+							"query": o.customQueryParam,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (o *OpenSearchLogProxy) getSearchRequestFromQuery(query map[string]interface{}) (*opensearchapi.SearchReq, error) {
+	var buf bytes.Buffer
+	if err := json.NewEncoder(&buf).Encode(query); err != nil {
+		return nil, errors.Wrap(err, "Failed to encode query")
+	}
+
+	return &opensearchapi.SearchReq{
+		Body: &buf,
+	}, nil
+}
+
+func (o *OpenSearchLogProxy) addMustClause(query map[string]interface{}, clauseType, field, value string) {
+	boolQuery := query["query"].(map[string]interface{})["bool"].(map[string]interface{})
+
+	mustClause, ok := boolQuery["must"].([]interface{})
+	if !ok {
+		mustClause = []interface{}{}
+	}
+
+	boolQuery["must"] = append(mustClause, map[string]interface{}{
+		clauseType: map[string]interface{}{
+			field: value,
+		},
+	})
+}
+
+func (o *OpenSearchLogProxy) addTermsFilter(query map[string]interface{}, field string, values []string) {
+	if len(values) == 0 {
+		return
+	}
+
+	boolQuery := query["query"].(map[string]interface{})["bool"].(map[string]interface{})
+
+	// Ensure "filter" exists and is a slice
+	filters, ok := boolQuery["filter"].([]interface{})
+	if !ok {
+		filters = []interface{}{}
+	}
+
+	// Append the new terms filter
+	filters = append(filters, map[string]interface{}{
+		"terms": map[string]interface{}{
+			field: values,
+		},
+	})
+
+	// Reassign the updated filters slice
+	boolQuery["filter"] = filters
+}
+
+func (o *OpenSearchLogProxy) addTimeFilter(query map[string]interface{}, timeFilter *platform.TimeFilter) {
```

Comment:
Same for most of the filters, we should align it with elastic where possible and use marhalling, to avoid issues with duplicating logic.